File: Api/DataLayer.Dados/Entities/Advogado.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace DataLayer.Dados.Entities
{
    public class Advogado
    {
        public int IdAdvogado { get; set; }
        public string Nome { get; set; }
        public string InscricaoOAB { get; set; }
        public string CodSegurancaOAB { get; set; }
        public string ExpedicaoOAB { get; set; }
        public string Foto { get; set; }
        public DateTime DataCadastro { get; set; }

        public string CEP { get; set; }
        public string Bairro  { get; set; }
        public string  Logradouro { get; set; }
        public string Cidade   { get; set; }
        public string  UF { get; set; }
        public string Biografia { get; set; }
        public string Complemento { get; set; }
        public string Especializacao { get; set; }
    }
}

File: Web/src/components/advogado/advogado.ts
export class Advogado {
    IdAvogado: number;
    Nome: string;
    InscricaoOAB: string;
    CodSegurancaOAB: string;
    ExpedicaoOAB: string;
    Foto: string;
    Cep:string;
    Logradouro:string;
    Cidade:string;
    Uf:string;
    Bairro:string;
    Complemento:string;
    Especializacao:string;
    Biografia:string

    constructor(idadvogado?: number, nome?: string, inscricaooab?: string, 
        codseguranca?: string, expedicaoab?: string, foto?: string, cep?:string,
         logradouro?:string,cidade?:string,uf?:string, bairro?:string,complemento?:string,especializacao?:string,biografia?:string){

        this.IdAvogado = idadvogado;
        this.Nome = nome;
        this.InscricaoOAB = inscricaooab;
        this.CodSegurancaOAB = codseguranca;
        this.ExpedicaoOAB = expedicaoab;
        this.Foto = foto;
        this.Cep = cep;
        this.Logradouro = logradouro;
        this.Cidade = cidade;
        this.Uf = uf; 
        this.Bairro =bairro; 
        this.Complemento = complemento;
        this.Especializacao=especializacao;
        this.Biografia=biografia;
    }
}